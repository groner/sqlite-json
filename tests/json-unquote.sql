-- Should be SQL NULL
SELECT JSON_UNQUOTE('null') AS 'null';
SELECT JSON_UNQUOTE('null') IS NULL AS 'null IS NULL';
-- Should be null
SELECT JSON_UNQUOTE('"null"') AS 'null';
SELECT JSON_UNQUOTE('"null"') IS NOT NULL AS 'null IS NOT NULL';
-- Should be foo
SELECT JSON_UNQUOTE('"foo"') AS 'foo';
-- Should be 123
SELECT JSON_UNQUOTE('123') AS '123';
-- Should reject with an error
SELECT JSON_UNQUOTE('[1,2,3]') AS '???';
