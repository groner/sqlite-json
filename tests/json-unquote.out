-- Should be SQL NULL
SELECT JSON_UNQUOTE('null') AS 'null';
null      
----------
NULL      
SELECT JSON_UNQUOTE('null') IS NULL AS 'null IS NULL';
null IS NULL
------------
1           
-- Should be null
SELECT JSON_UNQUOTE('"null"') AS 'null';
null      
----------
null      
SELECT JSON_UNQUOTE('"null"') IS NOT NULL AS 'null IS NOT NULL';
null IS NOT NULL
----------------
1               
-- Should be foo
SELECT JSON_UNQUOTE('"foo"') AS 'foo';
foo       
----------
foo       
-- Should be 123
SELECT JSON_UNQUOTE('123') AS '123';
123       
----------
123       
-- Should reject with an error
Error: near line 12: cannot unquote object or array
SELECT JSON_UNQUOTE('[1,2,3]') AS '???';
